src/views/AddItem.jsx

```diff
@@ -1,3 +1,101 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+
+export function AddItem({ listToken }) {
+	const [daysUntilNextPurchase, setTimeFrame] = useState('7');
+	const [itemName, setItem] = useState('');
+	const [itemData, setData] = useState({});
+	const [error, setError] = useState(false);
+	const [success, setSuccess] = useState(false);
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		setData({ itemName, daysUntilNextPurchase });
+		try {
+			await addItem(listToken, { itemName, daysUntilNextPurchase });
+			setError(false);
+			setSuccess(true);
+		} catch (err) {
+			console.log(err.message);
+			setError(true);
+		}
+	};
+
+	const handleName = (e) => {
+		setSuccess(false);
+
+		setItem(e.target.value);
+	};
+
+	const handleTime = (e) => {
+		setTimeFrame(e.target.value);
+		setSuccess(false);
+	};
+	// TODO: require input for item name - done [V]
+	return (
+		<div>
+			<form onSubmit={handleSubmit}>
+				<div className="item-name">
+					<label htmlFor="item-name">
+						Item Name:{' '}
+						<input
+							required
+							value={itemName}
+							type="text"
+							name="item-name"
+							id="item-name"
+							placeholder="name of item"
+							onChange={handleName}
+						/>
+					</label>
+				</div>
+				<div>
+					<fieldset className="timeframe">
+						<legend className="legend">
+							How soon will you buy this again?
+						</legend>
+						<label htmlFor="soon">
+							<input
+								type="radio"
+								value="7"
+								checked={daysUntilNextPurchase === '7'}
```

Comment:
I'm curious what this checked={daysUntilNextPurchase === '7'} is doing. It's such a good solution (I removed it, and just left 'checked' as the default, and the behavior became quite buggie). I'd love to understand if you have a chance to explain!

Reply:
I believe the checked is necessary so we can run our getFutureDate function, which takes the daysUntilNextPurchased as the offset. What I am curious about though is the strict equals (===) comparing a string of '7' vs a number 7. I believe the strict equals does not do any type conversation, so will that affect anything? Meaning, do we set our timeframe state as a number 7, and then have this check read as "daysUntilNextPurchase == '7'" or "daysUntilNextPurchase === 7"??

Reply:
Yep, so we forgot to change it from a string to a number here as well! Thank you both for some sharp eyes